### PR TITLE
Adds hmac_sha256

### PR DESCRIPTION
**Pull request by bobjflong**
We would like to use Shopify proxies to generate authentication codes as a security feature.

### CODE REVIEW

**Comment by fw42** [comment]
I don't think there's many use-cases for this... But I also have no reason to say no to this, really..

@pushrax @dylanahsmith thoughts?

File: lib/liquid/standardfilters.rb [hunk 1]
```diff
@@ -335,6 +336,11 @@ def floor(input)
       raise Liquid::FloatDomainError, e.message
     end
 
+    def hmac_sha256(input, secret)
+      digest = OpenSSL::Digest.new('sha256')
+      OpenSSL::HMAC.hexdigest(digest, secret, input)
```
Comment (pushrax) [inline lib/liquid/standardfilters.rb]: Does this handle the case where the input or secret is nil? Usually in Liquid we allow nil values in filters and return nil.

**Comment by dylanahsmith** [comment]
I see that we have md5 and sha1 filters in Shopify, but nothing like that in the liquid standard filters.  Do you think this is likely useful outside of Shopify?  Or should we just have this as a filter in Shopify?

**Comment by pushrax** [comment]
Interesting idea. Also don't really have a reason why this shouldn't be allowed. I think it might make more sense to include it directly in Shopify though.

It sounds like a fun challenge to implement this in pure Liquid!

**Comment by nickpearson** [comment]
I have implemented md5 and sha1 as Liquid filters and use them fairly often, so I can see where this filter would be useful.

File: lib/liquid/standardfilters.rb (hunk 1 above)
Comment (bobjflong) [inline lib/liquid/standardfilters.rb]: added a nil check

**Comment by bobjflong** [comment]
I wasn't aware there was a division between Shopify and Liquid like that. Our use case involves Shopify proxies. I wouldn't be well placed to suggest where the change should live.

**Comment by bobjflong** [comment]
Anything I could do here? Something like this is the last piece of the puzzle for the Intercom integration. Is there someone in Shopify I should contact?

**Comment by fw42** [comment]
I will push this to Shopify core. Will keep you updated.

**Comment by fw42** [comment]
Thanks for your contribution!

**Comment by fw42** [comment]
Just added the following filters to Shopify:
```
  def sha256(input)
    Digest::SHA256.hexdigest(input.to_s)
  end

  def hmac_sha1(input, secret)
    OpenSSL::HMAC.hexdigest('sha1', secret.to_s, input.to_s)
  end

  def hmac_sha256(input, secret)
    OpenSSL::HMAC.hexdigest('sha256', secret.to_s, input.to_s)
  end
```

**Pull request closed without merging**